addons/auth_signup_recaptcha/static/src/signup.js

```diff
@@ -0,0 +1,43 @@
+/** @odoo-module **/
+
+import publicWidget from "@web/legacy/js/public/public_widget";
+import { ReCaptcha } from "@google_recaptcha/js/recaptcha";
+
+const CaptchaFunctionality = {
+    events: {
+        submit: "_onSubmit",
+    },
+
+    init() {
+        this._super(...arguments);
+        this._recaptcha = new ReCaptcha();
+    },
+
+    async willStart() {
+        return this._recaptcha.loadLibs();
+    },
+
+    _onSubmit(event) {
+        if (!this.$el.find("input[name='recaptcha_token_response']").length) {
+            event.preventDefault();
+            this._recaptcha.getToken(this.token_name).then((tokenCaptcha) => {
+                this.$el.append(
+                    `<input name="recaptcha_token_response" type="hidden" value="${tokenCaptcha.token}"/>`,
```

Comment:
I don’t like when we construct html by hand like this (potential of xss) instead of `setAttribute`

Reply:
> I think the proper solution would be turning this into vanilla JS 

You will need to do it one day, or even worse, in owl